src/resolve/resolved-dns-transaction.c

```diff
@@ -1124,6 +1125,33 @@ void dns_transaction_process_reply(DnsTransaction *t, DnsPacket *p, bool encrypt
 
                 if (!t->bypass &&
                     IN_SET(DNS_PACKET_RCODE(p), DNS_RCODE_FORMERR, DNS_RCODE_SERVFAIL, DNS_RCODE_NOTIMP)) {
+                        /* If the server has replied with detailed error data, using a degraded feature set
+                         * will likely not help anyone. Examine the detailed error to determine the best
+                         * course of action. */
+                        r = dns_packet_ede_rcode(p, &ede_rcode);
+                        if (r < 0)
+                                goto fail;
+                        if (r > 0 && DNS_PACKET_RCODE(p) == DNS_RCODE_SERVFAIL) {
+                                /* These codes are related to DNSSEC configuration errors. If accurate,
+                                 * this is the domain operator's problem, and retrying won't help. */
+                                if (ede_rcode_is_dnssec(ede_rcode)) {
+                                        dns_transaction_complete(t, DNS_TRANSACTION_DNSSEC_FAILED);
```

Comment:
can you add a debug message here explaining the situation, i.e. why and what is done